jax/_src/api.py

```diff
@@ -2904,6 +2904,37 @@ def transposed_fun(const, out_cotangent):
   return Partial(transposed_fun, const)
 
 
+@overload
+def make_jaxpr(fun: Callable,
+               static_argnums: Union[int, Iterable[int]] = (),
+               axis_env: Optional[Sequence[Tuple[AxisName, int]]] = None,
+               return_shape: Literal[False] = False,
+               abstracted_axes: Optional[Any] = None,
+               ) -> Callable[..., core.ClosedJaxpr]:
+  ...
+
+
+@overload
+def make_jaxpr(fun: Callable,
+               static_argnums: Union[int, Iterable[int]] = (),
+               axis_env: Optional[Sequence[Tuple[AxisName, int]]] = None,
+               return_shape: Literal[True] = True,
```

Comment:
I don't think this is valid to have different default arguments for `return_shape` here and above.